### ISSUE TITLE
Build Your Own Value Stock Screener in Python (PE<15, PB<2, RSI<30)

### ISSUE THREAD

**Issue by jackyZhangtt** [jackyZhangtt/gitmoney#6]
**Title:** Build Your Own Value Stock Screener in Python (PE<15, PB<2, RSI<30)

**By: A Chief Officer with 15 years at sea, now navigating the markets**

---

### Introduction

I’ve spent 15 years on the bridge of cruise ships, mostly with Viking. You learn to read the weather, the currents, and the crew. But when the ship is docked and the engines are quiet, I sit in my cabin with a laptop and a cup of black coffee. That’s when I trade.

I’m not a Wall Street guy. I’m a sailor who learned that **value investing** works like a good passage plan: you need a clear destination, a margin of safety, and the discipline to stick to it when the wind shifts.

A few years ago, I got tired of manually scanning lists of stocks. I wanted a **python stock screener** that could filter for three things I trust: low valuation (PE<15, PB<2) and technical oversold conditions (RSI<30). That’s it. No hype, no meme stocks. Just a systematic approach to finding **oversold stocks** with a margin of safety.

Here’s how I built it. And yes, it runs on my old ThinkPad in the crew mess.

---

### 1. Why These Three Filters? (The Captain’s Logic)

In shipping, we use the "three-minute rule" for emergencies: assess, act, communicate. In investing, I use a three-filter rule: **value, safety, and timing**.

- **PE < 15**: Price-to-Earnings ratio below 15. This is my "is it cheap?" filter. In shipping, a vessel at a 15x earnings multiple is expensive. I want a workhorse, not a trophy. Historically, PE below 15 has been a decent entry point for long-term returns.

- **PB < 2**: Price-to-Book ratio below 2. This is my "margin of safety" filter. If the company goes under, what’s left? A PB below 2 means you’re paying less than twice the net asset value. In shipping, that’s like buying a ship at 80% of its scrap value.

- **RSI < 30**: Relative Strength Index below 30. This is my "timing" filter. RSI measures recent price momentum. Below 30 means the stock is oversold. It’s like a storm passing—the ship is down, but the hull is sound.

I don’t chase growth. I chase **oversold stocks** with solid fundamentals. The Python screener does the heavy lifting.

---

### 2. Building the Screener: Code That Works at Sea

I use Python with `yfinance` for data and `pandas` for filtering. No fancy APIs, no cloud subscriptions. Here’s the core logic (simplified for this article):

```python
import yfinance as yf
import pandas as pd

def get_stock_data(ticker):
    stock = yf.Ticker(ticker)
    info = stock.info
    hist = stock.history(period="6mo")
    
    # Calculate RSI
    delta = hist['Close'].diff()
    gain = (delta.where(delta > 0, 0)).rolling(window=14).mean()
    loss = (-delta.where(delta < 0, 0)).rolling(window=14).mean()
    rs = gain / loss
    rsi = 100 - (100 / (1 + rs))
    
    return {
        'PE': info.get('trailingPE'),
        'PB': info.get('priceToBook'),
        'RSI': rsi.iloc[-1]
    }

# Example: screening a list of stocks
tickers = ['AAPL', 'MSFT', 'XOM', 'JPM', 'WMT']
screened = []

for t in tickers:
    data = get_stock_data(t)
    if data['PE'] < 15 and data['PB'] < 2 and data['RSI'] < 30:
        screened.append(t)

print(f"Oversold value stocks: {screened}")
```

I run this script every Sunday evening before the US market opens. It takes about 10 minutes. I then manually check the top 5 results for earnings quality and debt levels. No AI, no black boxes. Just a **python stock screener** that works like a reliable ECDIS.

---

### 3. Real Results (and One Embarrassing Mistake)

In March 2023, my screener flagged **Ford (F)**. PE was 8.5, PB was 0.9, RSI was 28. I bought 200 shares at $11.20. Three months later, it hit $15. That’s a 34% gain. Not bad for a boring stock.

But I also learned a hard lesson. In October 2022, the screener flagged **a regional bank** (I won’t name it). PE was 6, PB was 0.7, RSI was 25. I bought in. Two months later, the bank reported a massive loan loss provision. The stock dropped 40%. I sold at a loss.

The problem? My screener didn’t check for **debt quality** or **regulatory risks**. Now I always do a 5-minute manual check: debt-to-equity ratio, insider selling, and recent news. The **python stock screener** is the first mate, not the captain.

---

### Conclusion

I’m still a sailor. I still stand watch. But I’ve learned that **value investing** doesn’t require a Bloomberg terminal or a CFA. It requires a clear framework, a little Python, and the discipline to follow your own rules.

If you’re a maritime professional like me—busy, practical, and skeptical of hype—try building your own **python stock screener**. Start with PE<15, PB<2, and RSI<30. You’ll be surprised how many **oversold stocks** are hiding in plain sight, waiting for a calm sea and a steady hand.

---

*Next time, I’ll share how I backtest this screener over 10 years of data. But first, I have a ship to sail.*

---

**[CTA placeholder: Add your signup link or download offer here]**

---
*About the Author:* Chief Officer by day, value investor by night. I run a **weekly value stock watchlist** with PE<15, PB<2, RSI<30 filters. 3-year track record targeting 30% annual returns.

📈 Get the weekly watchlist: [GitHub Sponsors](https://github.com/sponsors/jackyZhangtt) ($20/mo)
🐍 Python screener code available in this repo.
